core/src/utils/test/ready.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { componentOnReady } from '../helpers';
+
+describe('componentOnReady()', () => {
+  it('should correctly call callback for a custom element', async (done) => {
+    customElements.define('hello-world', class extends HTMLElement {
+      constructor() {
+        super();
+      }
+    });
+
+    const component = document.createElement('hello-world');
+    componentOnReady(component, () => done())
+  });
+
+  it('should correctly call callback for a lazy loaded component', async (done) => {
+    customElements.define('hello-world', class extends HTMLElement {
+      constructor() {
+        super();
+      }
+
+      componentOnReady() {
+        return new Promise((resolve) => {
```

Comment:
Maybe there should be a check that `componentOnReady()` is called too ?
(The current test would pass even if not)